### DISCUSSION TITLE
buff manager (wip)

### DISCUSSION THREAD

**Discussion by Eirev** [fedorovvl/tso_client#15] [Ideas]
что же это за что за "experimental buff manager (wip)"?
неужели возможность автоматического бафа строений на острове? 
думал когда-то о печальном отсутствии возможности бафнуть строения при помощи какой-нибудь рамки выделения, так что все строения, оказавшиеся в границах рамки, оказались под бафом (если это возможно)
неужели планируется что-то аналогичное?

**Comment by fedorovvl** [comment]
кусок баф менеджера доступен по нажатию F5
запись что и чем бафать похожа на запись макроса.. запускаете запись, бафаетесь, останавливаете запись
сохраняете шаблон, и в последующем по этому шаблону клиент бафает здания сам..
работать будет как у себя так и у друга
бафы из зведы будут браться сами.. если например 2 стака одинаковых бафов и один закончится то будет использован следующий
пока реализованы только шаблоны.. сам процесс бафа еще не реализован

**Reply by Eirev** [reply]
звучит отлично, будем ждать релиза. 
есть пожелание иметь возможность поставить макрос на паузу, ведь в игре бывают лаги и чтобы в случае, если из-за лага что-то окажется не бафнуто, не приходилось запускать все заново.

**Reply by fedorovvl** [reply]
не надо ждать релиза.. он уже случился)

**Reply by Eirev** [reply]
все отлично работает, спасибо большое за труды!

**Comment by BasilMas** [comment]
двух и более стаков бафов уже давно нет (вроде бы), все стакается в один пул.
У меня такой вопрос - а если при записи макроса что-то пропустил, все по-новой записывать придется?

**Comment by fedorovvl** [comment]
можно как дозаписать так и удалить строки

**Comment by BasilMas** [comment]
я пока не нашел как это сделать

**Comment by fedorovvl** [comment]
может это потому что не доделано еще? :)

**Comment by BasilMas** [comment]
ок, меня смутило отсутствие слова "будет" после "можно".

**Comment by fedorovvl** [comment]
в новом релизе можно попробовать в работе

**Comment by skelgaard** [comment]
microsoft defender says there is virus in the latest release of the client.exe

**Comment by fedorovvl** [comment]
as always when i modify cs files =)

**Reply by skelgaard** [reply]
its the first time, my defender has given that error on the file

**Reply by fedorovvl** [reply]
ignore it or build from source by yourself

**Reply by skelgaard** [reply]
i can, but the reason for it complaining, should be fixed instead... would be alot easier for most ppl

**Reply by fedorovvl** [reply]
i dont know why microsoft defender sees "virus" in client.. i think its because changing main source files produce new exe checksum or smth.. 
for defender that file is completelly new "downloaded from internet" and it dont trust him =)

**Reply by skelgaard** [reply]
https://www.virustotal.com/gui/file/093351aff32d9ec4a35509c89f8bf4d5e56346650f57d8a4899130a9531f0fe9

**Reply by fedorovvl** [reply]
https://opentip.kaspersky.com/093351AFF32D9EC4A35509C89F8BF4D5E56346650F57D8A4899130A9531F0FE9/

**Reply by skelgaard** [reply]
yes only 3 detects a problem on my link, but thats still 3 too much ;)

**Reply by skelgaard** [reply]
i googled a little on this error
"In this case, the binary was detected as suspicious, probably because it was compiled in debug mode. When the same code is compiled in release mode, we do not detect this as suspicious.
can this be the case for the release ?

**Reply by fedorovvl** [reply]
nope, its release build

**Comment by Urvdmih** [comment]
Почему решили на F5 баф сделать? По порядку типа? По мне рефреш зон под эту клавишу напрашивался, обычно во многих программах эта клавиша по умолчанию как "обновить" идет. Кстати хорошо бы клавишу в скобках сразу писать рядом с текстом, например "Filter (F1)".  Ну или сделать минисправку внутри программы.

**Reply by fedorovvl** [reply]
да, просто по порядку. клавишу в скобках хорошая идея, в след версии обязательно сделаю

**Comment by Anprivate** [comment]
Вот теперь ещё сделать сброс буффов по расписанию и можно вообще в игру не заходить по несколько дней :)

**Reply by BasilMas** [reply]
> Вот теперь ещё сделать сброс буффов по расписанию и можно вообще в игру не заходить по несколько дней :)

сорри за офтоп, но ощущение, что игра для Вас не фан, а какая-то ненавистная голгофа, куда с ужасом приходится ходить как на опостылевшую работу и отсиживать по 8 часов от звонка до звонка(

**Reply by Anprivate** [reply]
Вне эвентов это примерно так и есть. После 70го уровня по большому счёту ничего нового уже не увидишь. Так что автоматизировать рутинные действия было бы очень удобно.

**Comment by Urvdmih** [comment]
Почему-то багает удаление строчки из записи бафов. Нажимаю на крестик и весь лист очищается.

**Reply by Urvdmih** [reply]
В общем багает почему-то на десятке. В 7ке проблемы нет, и дозапись работает нормально и строчки удаляются. Почему так не понятно.

**Comment by Urvdmih** [comment]
Еще попрошу, если возможно, сделать так, чтобы клиент запоминал папку, куда я сохраняю шаблон (template). А то каждый раз выбирать заново приходится, очень неудобно.

**Reply by fedorovvl** [reply]
невозможно.. я уже писал об этом

**Reply by Urvdmih** [reply]
Жаль. Ну ладно.

**Reply by BasilMas** [reply]
Как временное решение - создать в папке Мои документы  ярлык на нужную папку. Чтобы ярлык был наверху - создать папку типа _EM с отчерком, и поместить ярлык в неё. Такая папка всегда сверху списка

**Reply by Urvdmih** [reply]
> Как временное решение - создать в папке Мои документы ярлык на нужную папку. Чтобы ярлык был наверху - создать папку типа _EM с отчерком, и поместить ярлык в неё. Такая папка всегда сверху списка

Ярлык является файлом, а значит как его не называй, он будет после папок в конце. Если в папке документов куча других папок, то ярлык будет внизу под ними. Но идею подсказали интересную, можно попробовать сделать симлинк папку. Она вроде будет отображаться именно как папка.

**Comment by fedorovvl** [comment]
в последнем апдейте добавлены 2 типа бафов в записывалку. нужны ваши отзывы, лучше/хуже
с флагом --debug если бафф не записался то появится алерт сверху с его типом.. если уверены что такие бафы нужно записывать то пришлите мне этот тип

in last update two new buff types added to buff recorder, so theoretically it must record more buffs
i need some feedback
also with --debug flag when buff not recorded by some reason warning alert will appear with buff type. if you sure that this buff must be recorded then show me this type

**Reply by igorSof** [reply]
всё отлично, не хватает этого http://joxi.ru/gmvjZ4H0eo04ma
RemoveBuff

**Reply by igorSof** [reply]
вот еще здания бафают, тоже не хватает http://joxi.ru/bmoBQ4HlO4Xxry
AreaBuffLoveGardenLvl1

И не хватает Таймера, чтобы оно периодически бафалось, через определенное время

**Reply by igorSof** [reply]
еще переплетчик http://joxi.ru/gmvjZ4H0e4B6ma
BookbinderBuffLvl4

**Reply by AceOfSettlers** [reply]
@fedorovvl hello , this is the buff i was talking about 
![194153740-6186b4f7-4451-4796-8da0-e612642eddce](https://user-images.githubusercontent.com/114313240/194742637-dfd5810f-859d-4234-ae37-99f2c21e5aae.png)
![194153828-188b176f-f174-49b3-a269-013cd7bd7882](https://user-images.githubusercontent.com/114313240/194742645-e7b79495-f8ad-40a7-a4ed-3aef9930e45d.png)

**Reply by AceOfSettlers** [reply]
hello Everybody , in the new version of the client , the Remove buff can be recorded but can't be apply to delete the existing buff  , any clue @fedorovvl to resolve it in a new version of the client ? 
![196387255-83bbafdf-543d-4f49-9be3-c17c2afd865c](https://user-images.githubusercontent.com/114313240/197218746-9ae6474d-0ee8-4bff-b2ce-a83674fa585a.png)

**Reply by fedorovvl** [reply]
working on it. soon

**Reply by YouRQXXL** [reply]
Buffs from Halloween event:
Trick - HalloweenEvent_Horror
Treat - HalloweenEvent_Ghostbuster
